### PR TITLE
Raven::Instance should use its own context/config when creating Event

### LINKED ISSUE #597: Raven::Instance still uses global context and configuration
When creating a `Raven::Event`, it just calls `from_message` or `from_exception` and those use `Raven.context` and `Raven.configuration` (there appear to be some ways to override this behaviour, but `Raven::Instance` is not using them)

### PR DESCRIPTION
**Pull request by singpolyma-shopify**
Closes #597

### CODE REVIEW

File: lib/raven/event.rb [hunk 1]
```diff
@@ -39,20 +39,21 @@ def initialize(init = {})
...
       @server_os     = {} # TODO: contexts
       @runtime       = {} # TODO: contexts
       @tags          = {} # TODO: contexts
       @checksum      = nil
       @fingerprint   = nil
-      @environment   = @configuration.current_environment
 
       yield self if block_given?
 
+      @environment   = @configuration.current_environment
```
Comment (nateberkopec) [inline lib/raven/event.rb]: Why reorder this section?

Comment (nateberkopec) [inline lib/raven/event.rb]: Placing it after `yield self` is a problem.

Comment (singpolyma-shopify) [inline lib/raven/event.rb]: @nateberkopec I've pushed an alternate implementation that solves my issue without the re-ordering

**Comment by nateberkopec** [comment]
@singpolyma-shopify Can you see if #601 also meets your needs? I don't want to change method signatures where possible.

**Pull request closed without merging**